# Midterm 1

## FINM 36700 - 2023

### UChicago Financial Mathematics

* Mark Hendricks
* hendricks@uchicago.edu

# Instructions

## Please note the following:

Points
* The exam is 100 points.
* You have 120 minutes to complete the exam.
* For every minute late you submit the exam, you will lose one point.
Final Exam

Submission
* You will upload your solution to the `Midterm 1` assignment on Canvas, where you downloaded this. (Be sure to **submit** on Canvas, not just **save** on Canvas.
* Your submission should be readable, (the graders can understand your answers,) and it should **include all code used in your analysis in a file format that the code can be executed.** 

Rules
* The exam is open-material, closed-communication.
* You do not need to cite material from the course github repo--you are welcome to use the code posted there without citation.

Advice
* If you find any question to be unclear, state your interpretation and proceed. We will only answer questions of interpretation if there is a typo, error, etc.
* The exam will be graded for partial credit.

## Data

**All data files are found in the class github repo, in the `data` folder.**

This exam makes use of the following data files:
* `midterm_data_1.xlsx`

This file has sheets for...
* `info` - names of each stock ticker
* `excess returns` - weekly excess returns on several stocks
* `SPY` - weekly excess returns on SPY

Note the data is **weekly** so any annualizations should use `52` weeks in a year.

#### If useful
here is code to load in the data.

In [33]:
# FILEIN = '../data/midterm_1_data.xlsx'
# sheet_exrets = 'excess returns'
# sheet_spy = 'spy'

# retsx = pd.read_excel(FILEIN, sheet_name=sheet_exrets).set_index('date')
# spy = pd.read_excel(FILEIN, sheet_name=sheet_spy).set_index('date')

## Scoring

| Problem | Points |
|---------|--------|
| 1       | 20     |
| 2       | 35     |
| 3       | 30     |
| 4       | 15     |

### Each numbered question is worth 5 points.

### Notation
(Hidden LaTeX commands)

$$\newcommand{\mux}{\tilde{\boldsymbol{\mu}}}$$
$$\newcommand{\wtan}{\boldsymbol{\text{w}}^{\text{tan}}}$$
$$\newcommand{\wtarg}{\boldsymbol{\text{w}}^{\text{port}}}$$
$$\newcommand{\mutarg}{\tilde{\boldsymbol{\mu}}^{\text{port}}}$$
$$\newcommand{\wEW}{\boldsymbol{\text{w}}^{\text{EW}}}$$
$$\newcommand{\wRP}{\boldsymbol{\text{w}}^{\text{RP}}}$$
$$\newcommand{\wREG}{\boldsymbol{\text{w}}^{\text{REG}}}$$

# 1. Short Answer

### No Data Needed

These problem does not require any data file. Rather, analyze the situation conceptually, based on the information below. 

## 1

In what sense was ProShares `HDG` successful in hedging the `HFRI`, and in what sense was it unsuccessful in tracking the `HFRI`?

HDG is successful in hedging the HFRI when there is negative correlation between HDG and HFRI's return in the market. Also, when the alpha is large, the hedging process is successful. 

HDG is unseccessful in tracking the HFRI when it deviates from positions if they can get additional returns or reduce transaction cost. When the tracking error is high, it is unseccessful. When the alpha value is large, the tracking is unsuccessful. 

## 2

We discussed multiple ways of calculating Value-at-Risk (VaR). What are the tradeoffs between using the normal distribution formula versus a directly empirical approach?

Calculating VaR using normal distribution will gain statistical power for the result. For example, there are only 50 datapoints and we want to see the VaR at quantile 0.01. There is no such point that can be estimated in the sample, or the sample size is too limited. Therefore, assuming normal distribution gives the result in more statistical way. However, when the data is not normally distributed, it causes some problems because it overestimate or underestimate the value of VaR.

Calculating VaR using directly empirical approach is great when there is a huge dataset and the sample for quantile 0.05 and 0.01 is enough to estimate. When there are limited sample data to estimate, it loses statistical power in estimating VaR. 

## 3

Did we find that **TIPS** have been useful in expanding the mean-variance frontier in the past? Did we conclude they might be useful in the future? Explain.

Somewhat yes. TIPS has relatively low correlation with other assets. When TIP is involved in the mean-variance frontier calculation, the Sharpe Ratio is relatively higher (changes from 1.930 to 1.933). However, it does not changes significantly, which means that the inclusion of TIPS or not does not significantly affect the mean-variance frontier. It is very useful in the future, because when you expect TIPS to grow or downgrade in the future, it will have significant impact on the mean-variance frontier. For example, when expected excess return to TIPS is adjusted to be 0.0012 higher than what the historic sample shows, the sharpe ratio increases from 1.930 to 2.031 with lower volatility, and the weights are completely different. Dropping TIPS from the investment set barely impacts the weights or the resulting performance. Adjusting the mean of TIPS upward even just 1 standard error substantially impacts the allocations and moderately boosts the resulting performance. Based on just a mean-variance analysis, it seems one could reasonably go either way with TIPS as an alternate asset class. In the argument to keep it separate, there is more diversification between TIPS and bonds than between SPY and many other equity buckets Harvard has. On the other hand, TIPS mostly impact the allocation to domestic bonds and might be seen as another asset in that bucket.

## 4.

What aspect of the classic mean-variance optimization approach leads to extreme answers? How did regularization help with this issue?

mean-variance optimization approach leads to extreme answers because when the number of assets is really large, the correlation between assets are large. In order to minimize the volatility, it will long and short assets with high correlation with extremely large leveraging scale. Estimation Errors: One of the main challenges with mean-variance optimization is the sensitivity of the approach to the input estimates, especially expected returns, variances, and covariances of the assets. Small changes or errors in input estimates can lead to vastly different portfolio weights, and in many cases, extreme portfolio allocations (i.e., very large or very small weights, even negative weights in certain unconstrained optimizations. Inverse of the Covariance Matrix: The optimization process involves the inversion of the covariance matrix. If this matrix is nearly singular (i.e., has a determinant close to zero), or if it's ill-conditioned, it can lead to extreme asset weights. This problem is exacerbated when dealing with a large number of assets, some of which might be closely correlated. Regularization introduces a penalty on the magnitude of portfolio weights, especially when trying to estimate parameters in a statistical model. It helps in constraining or shrinking the estimates towards more stable solutions.Ridge (L2) and Lasso (L1) Regularization: These are the most common techniques used to introduce regularization in portfolio optimization. Ridge regularization tends to shrink the coefficients toward zero but doesn't set any of them exactly to zero. Lasso regularization can shrink some coefficients to exactly zero, effectively selecting a subset of assets for the portfolio. This can be useful in creating sparse portfolios. Regularization helps in producing more stable portfolio weights that are less sensitive to small changes in input estimates. Especially with L1 regularization, the resulting portfolios tend to be simpler, with fewer assets having non-zero weights, which can aid in interpretability and practicality (e.g., reduced transaction costs). Regularization helps in preventing overfitting to the sample data, which is particularly crucial when dealing with financial data that can be noisy. Bayesian Approaches: Another form of regularization comes from Bayesian methods, where prior beliefs about asset returns can be combined with observed data to create more stable posterior estimates. This acts as a form of regularization by pulling extreme estimates towards more reasonable values based on prior beliefs. In essence, regularization provides a tool to address the sensitivity and instability issues inherent in classic mean-variance optimization, allowing for more practical and robust portfolio solutions.

In [53]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt
import seaborn as sns
from statsmodels.regression.rolling import RollingOLS
from functools import partial
import sys
import math
pd.options.display.float_format = "{:,.4f}".format
from sklearn.linear_model import LinearRegression
import warnings
warnings.filterwarnings("ignore")
import os
from scipy.stats import norm

***

# 2. Allocation

Consider a mean-variance optimization of **excess** returns provided in `midterm_1_data.xlsx.`

## 1. 

Report the following **annualized** statistics:
* mean
* volatility
* Sharpe ratio

Which assets have the highest / lowest Sharpe ratios?

In [10]:
FILEIN = '/Users/yiningqu/Desktop/midterm_1_data.xlsx'

sheet_exrets = 'excess returns'
sheet_spy = 'spy'

retsx = pd.read_excel(FILEIN, sheet_name=sheet_exrets).set_index('date')
spy = pd.read_excel(FILEIN, sheet_name=sheet_spy).set_index('date')


In [16]:
annualized_mean = retsx.mean()*52
print(annualized_mean)

AAPL    0.3194
MSFT    0.2881
AMZN    0.2395
NVDA    0.6507
GOOGL   0.1933
TSLA    0.5697
XOM     0.1242
dtype: float64


In [17]:
annualized_std = retsx.std()* math.sqrt(52)
print(annualized_std)

AAPL    0.2839
MSFT    0.2402
AMZN    0.3104
NVDA    0.4681
GOOGL   0.2742
TSLA    0.6070
XOM     0.3116
dtype: float64


In [18]:
sharpe_ratio = annualized_mean / annualized_std
print(sharpe_ratio)

AAPL    1.1252
MSFT    1.1993
AMZN    0.7715
NVDA    1.3900
GOOGL   0.7050
TSLA    0.9386
XOM     0.3986
dtype: float64


In [20]:
sr_array = np.array(sharpe_ratio)
sr_max = retsx.columns[np.argmax(sr_array)]
sr_min = retsx.columns[np.argmin(sr_array)]
print('Asset class with highest sharpe ratio is ', sr_max)
print('Asset class with lowest sharpe ratio is ', sr_min)

Asset class with highest sharpe ratio is  NVDA
Asset class with lowest sharpe ratio is  XOM


## 2.

Report the weights of the tangency portfolio.

Also report the Sharpe ratio achieved by the tangency portfolio over this sample.

In [23]:
cov_matrix = retsx.cov()
inv_cov = np.linalg.inv(cov_matrix)
mean = np.array(annualized_mean)
n_1 = np.transpose(np.ones(7))
w_tan = (1/(n_1@inv_cov@mean))*(inv_cov@mean)
w_tan_summary = pd.DataFrame(np.array([w_tan,sr_array]).T, columns= ["Tangent Weight","Sharpe Ratio"], index= retsx.columns)
w_tan_summary

Tangent Weight  Sharpe Ratio
AAPL           0.3226        1.1252
MSFT           0.7875        1.1993
AMZN          -0.2286        0.7715
NVDA           0.4960        1.3900
GOOGL         -0.5027        0.7050
TSLA           0.1060        0.9386
XOM            0.0193        0.3986

## 3.

* What weight is given to the asset with the lowest Sharpe ratio?
* What Sharpe ratio does the lowest (most negative) weight asset have?

Explain. Support your answer with evidence.

In [25]:
w_min_sr = w_tan_summary.loc[w_tan_summary['Sharpe Ratio'].idxmin()]['Tangent Weight']
sr_min_w = w_tan_summary.loc[w_tan_summary['Tangent Weight'].idxmin()]['Sharpe Ratio']


print('The weight of the asset with the lowest Sharpe ratio is ', w_min_sr)
print('Sharpe ratio of lowest (most negative) weight asset is ', sr_min_w)

The weight of the asset with the lowest Sharpe ratio is  0.019256607953705398
Sharpe ratio of lowest (most negative) weight asset is  0.705019854216037


## 4.

Let's examine the out-of-sample performance.

Calculate and report the following three allocations using only data through the end of 2022:
* tangency portfolio
* equally weighted portfolio
* a regularized approach, with a new formula shown below

where
$$\wEW_i = \frac{1}{n}$$

$$\wREG \sim \widehat{\Sigma}^{-1}\mux$$

$$\widehat{\Sigma} = \frac{\Sigma + \boldsymbol{2}\,\Sigma_D}{\boldsymbol{3}}$$
where $\Sigma_D$ denotes a *diagonal* matrix of the security variances, with zeros in the off-diagonals.

In [96]:
excess_returns_IS = retsx[:'2022'].reset_index()
excess_returns_OOS = retsx['2023':].reset_index()

excess_returns_in_sample = excess_returns_IS[excess_returns_IS.columns[1:]] 
excess_returns_out_of_sample = excess_returns_OOS[excess_returns_OOS.columns[1:]]

In [97]:
def summary_statistics_annualized(returns, annual_factor = 52):
    """This functions returns the summary statistics for the input total/excess returns passed
    into the function"""
    
    summary_statistics = pd.DataFrame(index=returns.columns)
    summary_statistics['Mean'] = returns.mean() * annual_factor
    summary_statistics['Vol'] = returns.std() * np.sqrt(annual_factor)
    summary_statistics['Sharpe'] = (returns.mean() / returns.std()) * np.sqrt(annual_factor)
    summary_statistics['Min'] = returns.min()
    summary_statistics['Max'] = returns.max()
    summary_statistics['Skewness'] = returns.skew()
    summary_statistics['Excess Kurtosis'] = returns.kurtosis()
    summary_statistics['VaR (0.05)'] = returns.quantile(.05, axis = 0)
    summary_statistics['CVaR (0.05)'] = returns[returns <= returns.quantile(.05, axis = 0)].mean()
    
    return summary_statistics


wts_IS = pd.DataFrame(index = excess_returns_IS.columns[1:], columns = ['tangency','tangency with TIPS dropped',
                                                                      'tangency with TIPS adjusted','equal weights',
                                                                      'risk parity','regularized'])
summary_statistics_annualized(excess_returns_out_of_sample @ wts_IS)

Mean  Vol Sharpe   Min   Max Skewness  \
tangency                     NaN  NaN    NaN  None  None      NaN   
tangency with TIPS dropped   NaN  NaN    NaN  None  None      NaN   
tangency with TIPS adjusted  NaN  NaN    NaN  None  None      NaN   
equal weights                NaN  NaN    NaN  None  None      NaN   
risk parity                  NaN  NaN    NaN  None  None      NaN   
regularized                  NaN  NaN    NaN  None  None      NaN   

                            Excess Kurtosis  VaR (0.05) CVaR (0.05)  
tangency                                NaN         NaN         NaN  
tangency with TIPS dropped              NaN         NaN         NaN  
tangency with TIPS adjusted             NaN         NaN         NaN  
equal weights                           NaN         NaN         NaN  
risk parity                             NaN         NaN         NaN  
regularized                             NaN         NaN         NaN

In [101]:
target_mean = 1
#Equally-weighted (EW)
ew = pd.DataFrame()
weight1 = np.full((retsx.shape[1],1), 1/retsx.shape[1])
#np.full((number of rows, number of columns),each entry value)
mean = weight1.T @ np.array(retsx.mean() * 12)

variance = weight1.T @ retsx.cov() @ weight1

vol = math.sqrt(variance.iloc[0, 0]) *math.sqrt(12)

sharpe_ratio = mean / vol

rescale1 = target_mean / mean

ew['mean'] = mean * rescale1
ew['volatility'] = vol * rescale1
ew['sharpe ratio'] = sharpe_ratio
ew.index = ['Equally weighted']
ew

mean  volatility  sharpe ratio
Equally weighted 1.0000      1.5924        0.6280

In [102]:

#Regularized (REG)
reg = pd.DataFrame()

cov_matrix = np.array(retsx.cov()) 
sigma_d = np.diag(np.diag(cov_matrix))
matrix = (cov_matrix + sigma_d)/2
weight3 = np.linalg.inv(matrix) @ np.array(retsx.mean() * 12)
scale = 1/weight3.sum()
weight_reg = (weight3 * scale)

mean = weight_reg.T @ np.array(retsx.mean() * 12) 

variance = weight_reg.T @ retsx.cov() @ weight_reg
vol = math.sqrt(variance) *math.sqrt(12)

sharpe_ratio = mean / vol

rescale2 = target_mean / mean

reg['mean'] = [mean * rescale2]
reg['volatility'] = [vol * rescale2]
reg['sharpe ratio'] = [sharpe_ratio]
reg.index = ['Regularized']
reg




mean  volatility  sharpe ratio
Regularized 1.0000      1.4108        0.7088

## 5.

Report the out-of-sample (2023) performance of all three portfolios in terms of annualized mean, vol, and Sharpe.

## 6.

Imagine just for this problem that this data is for **total** returns, not excess returns.

Report the weights of the global-minimum-variance portfolio.

In [109]:
inv_cov = np.linalg.inv(cov_matrix)  # Assuming cov_matrix is your covariance matrix
ones_vector = np.ones(inv_cov.shape[0])
global_min_var_weights = inv_cov @ ones_vector / (ones_vector.T @ inv_cov @ ones_vector)
global_min_var = pd.DataFrame(global_min_var_weights, index=retsx.columns, columns=["weight"])
global_min_var

weight
AAPL   0.2062
MSFT   0.4912
AMZN   0.1609
NVDA  -0.1192
GOOGL  0.0114
TSLA  -0.0469
XOM    0.2964

## 7.

To target a mean return of 0.005%, would you be long or short this global minimum variance portfolio?

I would long this global minimum variance portfolio 

***

# 3. Performance

## 1. 

Report the following performance metrics of excess returns for Tesla (`TSLA`).
* skewness
* kurtosis

You are not annualizing any of these stats.

What do these metrics indicate about the nature of the returns?

In [30]:
skewness = retsx["TSLA"].skew()
kurtosis = retsx["TSLA"].kurtosis()
print(f'Telsa skewness is: {skewness}')
print(f'Telsa kurtosis is: {kurtosis}')

Telsa skewness is: 0.44145458268186166
Telsa kurtosis is: 1.5273762473269983


Skewness of tesla is positive. Skewness measures the degree of asymmetry or lack of symmetry in the distribution of a dataset. It quantifies the direction and amount of skew (departure from horizontal symmetry). Skewness is positive means that it has high probability of small losses and with smaller probability with big returns.  positive skewness indicates a distribution that is skewed to the right (or has a right tail), meaning that the right tail is longer than the left tail. Here, the mean is typically greater than the median.

Kurtosis measures the "tailedness" of a distribution, that is, the sharpness of the peak and the heaviness of the tails. It quantifies whether the data has heavy or light tails relative to a normal distribution. Kurtosis is less than 3 means that the distribution has thinner tails and a more rounded peak than the normal distribution. It suggests a lower occurrence of extreme values than expected.

## 2. 

Report the maximum drawdown for `TSLA` over the sample.
* Ignore that your data is in excess returns rather than total returns.
* Simply proceed with the excess return data for this calculation.

In [38]:
index = 1000*(1+retsx).cumprod()
previous_peaks = index.cummax()
drawdowns = ((index - previous_peaks)/previous_peaks)
max_drawdown = drawdowns.min()  
maxdrawdown = pd.DataFrame(max_drawdown, columns = ["Maximum drawdown"])
maxdrawdown

Maximum drawdown
AAPL            -0.3721
MSFT            -0.2995
AMZN            -0.4681
NVDA            -0.5923
GOOGL           -0.3483
TSLA            -0.6822
XOM             -0.6714

## 3.

For `TSLA`, calculate the following metrics, relative to `SPY`:
* market beta
* alpha
* sortino ratio

Annualize alpha and sortino ratio.

In [51]:
def regression(regressor, regressand,  period=52, multiple=False, alpha=True):
    X = sm.tools.add_constant(regressor) if alpha else regressor   #加constant to regression model

    model = sm.OLS(regressand, X, missing='drop').fit()  #run regression

    if alpha:
        if multiple==False:
            beta = model.params.iloc[1]
            alpha = model.params.loc['const'] *52

    else:   
        beta = model.params
    
    sortino_ratio = retsx['TSLA'].mean() / retsx['TSLA'][retsx['TSLA'] < 0].std() * math.sqrt(52)
    
    return beta, alpha, sortino_ratio #如果没有alpha则没有information ratio


beta,alpha,sortino_ratio  = regression(spy, retsx['TSLA'])
data = {
    "market beta": [beta],
    "alpha": [alpha],
    "sortino ratio": [sortino_ratio]
}
TSLA = pd.DataFrame(data, index=["Tesla"])
TSLA

market beta  alpha  sortino ratio
Tesla       1.7768 0.3095         1.6423

## 4.

Continuing with `TSLA`, calculate the full-sample, 5th-percentile CVaR.
* Use the `normal` formula, assuming mean returns are zero.
* Use the full-sample volatility.

Use the entire sample to calculate a single CVaR number. 

In [57]:
vol = retsx['TSLA'].std() * math.sqrt(52)
CVaR = 0 - vol * norm.pdf(-1.65) / 0.05
CVaR

-1.241549307100958

## 5.

Now calculate the 5th-percentile, one-period ahead, **VaR** for `TSLA`.

Here, calculate the running series of VaR estimates.

Again, 
* use the normal formula, with mean zero.

But now, use the rolling volatility, based on 
* rolling window or $m=52$ weeks.

Report the final 5 values of your calculated VaR series.

In [65]:

def prob(mu, sigma, h):
    return norm.cdf(-np.sqrt(h)*mu/sigma)


VaR = retsx['TSLA'].shift(1).expanding(min_periods = 52).quantile(0.05)
VaR_mu = VaR.mean()
VaR_sigma = VaR.std()

probability = prob(mu=VaR_mu, sigma=VaR_sigma, h=5)
print('Probability of VaR smaller than 0 in the next 5 weeks: {:,.2%}'.
      format(probability))

Probability of VaR smaller than 0 in the next 5 weeks: 100.00%


## 6. 

Calculate the out-of-sample **hit ratio** for your VaR series reported in your previous answer.

In [66]:
hit_ratio = 5*probability / 5
print(f'hit ratio is {hit_ratio}')

hit ratio is 1.0


***

# 4. Hedging

## 1. 

Consider the following scenario: you are holding a \$100 million long position in `NVDA`. You wish to hedge the position using some combination of 
* `AAPL`
* `AMZN`
* `GOOGL`
* `MSFT`

Report the positions you would hold of those 4 securities for an optimal hedge.

Note:
* In the regression estimation, include an intercept.
* Use the full-sample regression. No need to worry about in-sample versus out-of-sample.

In [84]:
def regression(regressor, regressand, period=52, multiple=False, alpha=True):
    X = sm.tools.add_constant(regressor) if alpha else regressor   #加constant to regression model

    model = sm.OLS(regressand, X, missing='drop').fit()  #run regression

    if alpha:
        beta = model.params.iloc[1:]    #Question2只有一个factor SPY所以beta只有一个，但在其他时候会有别的regressor
        alpha = model.params.loc['const'] * period
    else:   
        beta = model.params
    
    tracking_er_vol = (model.resid.std() * np.sqrt(period)) 

    return beta, alpha, model.rsquared,tracking_er_vol #如果没有alpha则没有information ratio

beta, alpha, r_squared, tracking_error = regression(
        retsx[['AAPL','AMZN','GOOGL','MSFT']], retsx['NVDA'])
coefficients = pd.DataFrame(index= ['AAPL','AMZN','GOOGL','MSFT'], columns=['coefficient'])
coefficients['coefficient'] = beta
coefficients.loc['const'] =  alpha
coefficients

coefficient
AAPL        0.3417
AMZN        0.4173
GOOGL      -0.0078
MSFT        0.5879
const       0.2738

In [85]:
hedge = pd.DataFrame(index= ['AAPL','AMZN','GOOGL','MSFT'], columns=['Positions'])
hedge['Positions'] = 100000000 * beta
hedge

Positions
AAPL  34,168,648.9989
AMZN  41,725,985.8798
GOOGL   -784,795.2478
MSFT  58,789,673.0294

## 2.

How well does the hedge do? Cite a regression statistic to support your answer.

Also estimate the volatility of the basis, (epsilon.)

In [86]:
print(r_squared)
print(tracking_error)

0.4581682361155385
0.34456153441586174


The beta sizes are all very realistic, with none of the betas being above "1" or below "-1". We also don't have extreme long/short positions, which is good. The largest beta size is 0.5879 which is still can be acceptable. However, the R^2 is relatively low which means the hedge is less effective, as the independent variable(s) can explain a smaller proportion of the variability in the dependent variable. The tracking_error value is large, A large tracking error indicates that the hedge is effective, as the investors can trade on the portion of NVDA uncorrelated with four equities, and this portion is large. It does well in the hedging positions,because betas are realistic and tracking error is high. 

## 3.

Report the annualized intercept. By including this intercept, what are you assuming about the nature of the returns of `NVDA` as well as the returns of the hedging instruments?

In [87]:
coefficients = pd.DataFrame(index= ['AAPL','AMZN','GOOGL','MSFT'], columns=['coefficient'])
coefficients['coefficient'] = beta
coefficients.loc['const'] =  alpha
coefficients

coefficient
AAPL        0.3417
AMZN        0.4173
GOOGL      -0.0078
MSFT        0.5879
const       0.2738

holding the returns of the hedging instruments constant, NVDA is expected to outperform by 27.38% annually, based purely on the constant term. The intercept essentially captures any systematic performance of NVDA that isn't captured by its relationship with the hedging instruments. If the intercept were zero, it would imply that the returns of NVDA can be completely explained by its exposure to the hedging instruments, and there are no other sources of systematic return. A non-zero intercept implies that there's an additional return component for NVDA that isn't accounted for by its relationship with the hedging instruments. For example, GOOGL has nearly zero intercept which means returns of NVDA can be completely explained by its exposure to the GOOGL. 